app/scripts/services/open-submenu-service.js

```diff
@@ -0,0 +1,28 @@
+(function () {
+  'use strict';
+
+  angular.module('ramlEditorApp')
+    .service('openSubMenuService', function openSubMenuService(
+      $timeout,
+      $window
+    ) {
+
+      var openSubMenu = function (scope, subMenuName) {
```

Comment:
No need for the variable you can just:

```javascript:
function openSubMenu(){ ... code here ...}
```

And after you do that, you can slide the `return` all the way to the top of the module. That way in longer modules you can see the interface of the module, without going all the way to the bottom.